contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-epilogue.y.c

```diff
@@ -216,6 +216,22 @@ TsqlFunctionConvert(TypeName *typename, Node *arg, Node *style, bool try, int lo
 		 */
 		result = makeTypeCast(helperFuncCall, typename, location);
 	}
+
+	else if (strcmp(typename_string, "binary") == 0 || strcmp(typename_string, "varbinary") == 0)
```

Comment:
For strncmp we need to specify the length for comparison. 

for eg:
```else if (strncmp(typename_string, "binary", 6) == 0 || strncmp(typename_string, "varbinary", 9) == 0)```

 If "typename_string" is shorter than the comparison length, strncmp will still return 0 (equality) if the available characters match. For example, "bin" would match "binary". This may not be desirable for this usecase.